DependencyInjection/Compiler/LoggerChannelPass.php

```diff
@@ -113,6 +113,7 @@ protected function createLogger($channel, $loggerId, ContainerBuilder $container
                 $logger = new DefinitionDecorator('monolog.logger_prototype');
             }
 
+            $logger->setPublic(true);
```

Comment:
IMO, we should make them public only if they are channels requested to exist by the configuration of the bundle, not just by other services requesting a specific channel in a tag (as such case does not get the logger through `Container::get`

Reply:
In general, making things public I think is a dying solution. If you use AbstractController, you can’t even get public services. I’m not against this change, but more broadly, we need to use DI. And if that’s too hard in some cases, we need to make it easier.

In this case, the best solution might be to add a binding for all you services (so, under _defaults) for your custom channel - e.g. if you create “api”, then you setup a $apiLogger as a bind to this service. Then you can get it anywhere by using that as its argument name.

@nicolas-grekas sorry to bother you, but this is what I was talking to you about a few days ago: where private services are a pain, but by using binding (if we like that solution), you can solve it easily.